lib/doorkeeper/models/access_token_mixin.rb

```diff
@@ -160,6 +169,30 @@ def scopes_match?(token_scopes, param_scopes, app_scopes)
           )
       end
 
+      # Checks whether the token custom attribute values match the custom
+      # attributes from the parameters.
+      #
+      # @param token [Doorkeeper::AccessToken]
+      #   The access token whose custom attributes are being compared
+      #   to the param_attributes.
+      #
+      # @param param_attributes [Hash]
+      #   A hash of the attributes for which we want to determine whether
+      #   the token's custom attributes match.
+      #
+      # @return [Boolean] true if the token's custom attribute values
+      #   match those in the param_attributes, or if both are empty/blank.
+      #   False otherwise.
+      def custom_attributes_match?(token, param_attributes)
+        return true if param_attributes.nil?
```

Comment:
This is fine for now, but it will be an issue if we handle `/oauth/authorize?tenant_id=TENANT_ID&...`.

It would be a good idea to leave a comment I believe.